### DISCUSSION TITLE
try-ignore-catch and switch-case

### DISCUSSION THREAD

**Discussion by KevinKrueger** [dotnet/csharplang#7503] [Language Ideas]
### Idea 1:
```
try
{...}
ignore(NullReferenceException);
```

Optional: 
```
try
{...}
ignore(NullReferenceException, ArgumentException);
```

--------------------------------------------------------

### Idea 2:
```
try
{...}
ignore(NullReferenceException)
catch (Exception ex)
{...}
```

Optional:

```
try
{...}
ignore(ArgumentOutOfRangeException)
catch
{
    case NullReferenceException nrefex:
    case ArgumentException argex:
        ...
        break;
    default:
        break;
}
```

### Background behind the ideas:

Now and then there are cases where an exception is thrown but it is not really relevant for the further process.

In the code then mostly a catch with an empty block remains, where then after weeks it is not recognizable, why actually the catch block remains empty.

The "ignore(..)" should make recognizable that at this point this type of error should be ignored.

Within the ignore, one can then also optionally list several exceptions.  For example: ignore(NullReferenceException, ArgumentException);

The second idea is to perform concrete error handling in the catch-block analogous to the type-related switch-case. This way, multiple types of exceptions could be handled identically without having to write the code multiple times.

**Comment by HaloFour** [comment]
You can already ignore specific exceptions with empty `catch` clauses:

```cs
try
{
    // ...
}
catch (NullReferenceException) { }
catch (ArgumentException) { }
```

And you can switch over the exception type in a general `catch` clause as well:

```cs
try
{
    // ...
}
catch (ArgumentOutOfRangeException) { }
catch (Exception ex)
{
    switch (ex)
    {
        case NullReferenceException nrefex:
            // ...
            break;
        default:
            break;
    }
}
```

**Reply by KevinKrueger** [reply]
Yes, and that is exactly what is not beautiful.
Why exactly that is not really beautiful is explained above.

**Reply by HaloFour** [reply]
"Beauty" is subjective.  New forms of existing syntax aren't going to be added just to appeal to stylistic preferences unless they can also provide other value or solve problems in novel ways.

**Reply by bernd5** [reply]
You should add a comment which explains why you ignore the catched exception. E.g. NullRefernceException should never be ignored.
It would be reasonable while retrying network requests in unstable networks

**Reply by KevinKrueger** [reply]
I totally agreed with you, I see it exactly the same way.
I must apologize if it came across that way.

Please do not misunderstand but have you dealt with the problem?

Behind this "beautiful appearance" I have also thought of something, it is namely to mark that at this point one or more exception/s must be ignored. This has the advantage that you can see at a glance that something is ignored during error handling without having to create a new code block.

Empty code blocks are to be avoided. (Idea 1)

The lower problem prevents nesting. (Idea 2)

Nesting must be avoided.

**Reply by HaloFour** [reply]
Ignoring exceptions is to be avoided, to the extent that it shouldn't be supported by a language feature that appears to give tacit approval of such behavior.

Everything else is a matter of style, and the language isn't going to add a bunch of slightly different ways of doing the same thing because different people have different stylistic preferences.

**Reply by KevinKrueger** [reply]
I agree that ignoring exceptions in programming should be avoided, especially if there is no explicit language feature that supports it. This can help avoid unexpected bugs and ensure the robustness of our applications.

In terms of coding style, I believe that clarity and maintainability are critical in software development. However, in C# we can often use different approaches to accomplish the same task. This often depends on the requirements of the project and the preferences of the development team.

The use of stylistic exceptions can, in some cases, help make the code more readable or better convey certain concepts.

Ultimately, the choice between strict adherence to conventions and the use of exceptions is a matter of trade-off. In software development, we should always act on the principle of "safety first," but we should also have the flexibility to choose the best approach for the task at hand.

**Reply by HaloFour** [reply]
> I agree that ignoring exceptions in programming should be avoided, especially if there is no explicit language feature that supports it. 

Importantly, there _shouldn't_ be an explicit language feature to support something that is a bad practice.

Language features are incredibly expensive and the language team has been very clear that they're not interested in adding new syntax to accomplish the same thing you can already do today in order to appeal to a specific stylistic preference.  That's not a design philosophy that can scale, and the team is more interested in solving novel problems.

**Reply by KevinKrueger** [reply]
The idea should not be to introduce language features for bad practices, but rather to have the ability to flexibly and wisely choose between different approaches to write the best code for our needs.

**Reply by HaloFour** [reply]
... by introducing language features to encourage bad practices.

Language design is incredibly expensive, even for small features: https://learn.microsoft.com/en-us/archive/blogs/ericgu/minus-100-points

**Comment by KevinKrueger** [comment]
> You should add a comment which explains why you ignore the catched exception. E.g. NullRefernceException should never be ignored. 

All good, you are of course right 😂
Have I only taken to fill the ideas!
@bernd5
